src/impl/regexParser.js

```diff
@@ -189,6 +190,34 @@ function extractASCII(match) {
   return [result, FixedOffsetZone.utcInstance];
 }
 
+const sqlDateRegex = /^(\d{4})-(\d\d)-(\d\d)$/,
+  sqlTimeRegex = /^(\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/,
+  sqlDateTimeRegex = /^(\d{4})-(\d\d)-(\d\d) (\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/;
+
+function extractSqlDate(match) {
+  const [, yearStr, monthStr, dayStr] = match,
+    result = fromStrings(0, yearStr, monthStr, dayStr, 0, 0, 0);
+  return [result, FixedOffsetZone.utcInstance];
+}
+
+function extractSqlTime(match) {
+  const now = new Date(),
+    yearStr = now.getUTCFullYear().toString(),
+    monthStr = (now.getUTCMonth() + 1).toString(),
+    dayStr = now.getUTCDate().toString(),
+    [, hourStr, minuteStr, secondStr] = match,
+    result = fromStrings(0, yearStr, monthStr, dayStr, hourStr, minuteStr, secondStr);
```

Comment:
For this one, instead of defaulting the year, month, and date, don't use `fromStrings()` and just return the object with parsed hour, minute, and second. See `fromISOTime` for that. `fromObject` in the main DateTime file will handle defaulting those higher-order units.